src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -592,6 +611,12 @@ private void MakeExplicitParameterTypeInferences(BoundExpression argument, TypeW
                 {
                     ExactOrBoundsInference(kind, _extensions.GetTypeWithAnnotations(argument), target, ref useSiteInfo);
                 }
+                else if (IsUnfixedTypeParameter(target) && kind is ExactOrBoundsKind.LowerBound)
+                {
+                    var ordinal = ((TypeParameterSymbol)target.Type).Ordinal;
+                    var typeWithAnnotations = _extensions.GetTypeWithAnnotations(argument);
+                    _nullableAnnotationLowerBounds[ordinal] = _nullableAnnotationLowerBounds[ordinal].Join(typeWithAnnotations.NullableAnnotation);
```

Comment:
nit: Given that we're now only using `Join` and the starting values are `NotAnnotated`. I'm wonder if we should just store a boolean. I'll leave that for consideration for you and second reviewer. I don't feel strongly (seems equivalent).

Reply:
I also don't feel strongly about it. This did inspire some additional tests for 'null!' and 'null' in a disabled context, so thanks :)